### ISSUE TITLE
Representing Tree Like data structure in terraform

### ISSUE THREAD

**Issue by parweza** [hashicorp/terraform-plugin-sdk#112]
### Terraform Version

```
Terraform v0.11.7

```

### Representing Tree Like data structure in terraform
Hi, 
I am a terraform provider developer and was trying to model following (go representation of data) as terraform resource schema but am not able to do so (terraform validation goes in infinite recursion).  
Is it possible to model the following in terraform. if yes , Can you please point me some documentation? 
```hcl
type Model struct {

	FieldType ModelTypeEnum 

	
	FieldName *string


	IsArray *bool 

	Properties []Model 
}
```

**Comment by paultyng** [comment]
I believe the only way to accomplish this now in 0.12 is dynamic type support, linking this to #248

**Comment by paultyng** [comment]
This also may be solved by direct support in #261 .

Unfortunately, per Martin's comments above, I don't think this will be possible as TF core does not support any sort of recursive type mechanism.
